mcs/class/corlib/Test/System/ArraySortArgChecks.cs

```diff
@@ -311,88 +311,47 @@ public class ArraySortArgChecks {
 		}
 	
 		[Test]
-		public void Check_InvalidOperationException() {
-			object[] arr = new object[] {new SomeComparable (), new SomeIncomparable (), new SomeComparable ()};
+		public void Check_InvalidOperationException() 
+		{
+			Array arr = new object[] {new SomeComparable (), new SomeIncomparable (), new SomeComparable ()};
 	
-			try {
-				Array.Sort (arr);
-				Assert.Fail ("#1");
-			} catch (InvalidOperationException) {}
+			Array.Sort (arr);
```

Comment:
maybe use `Assert.DoesNotThrow (() => Array.Sort (arr));` so we get better failure traces when it does fail?